packages/commons/src/auth/jwt.ts

```diff
@@ -70,14 +71,17 @@ export const verifyJwtToken = async (
   jwtToken: string,
   config: JWTConfig,
   logger: Logger
-): Promise<boolean> => {
+): Promise<{
+  verified: boolean;
+  maybeDecoded?: JwtPayload | string;
+}> => {
```

Comment:
Suggestion: let's use a union type to model the two cases so we don't have an optional field.
```suggestion
): Promise<
{ verified: true; decoded: JwtPayload | string } | { verified: false }
> => {
```

Reply:
If you check `verified === true` then TS automatically infers the fact that decoded is defined, since this is a discriminated union - then you should be good to go. This approach excludes by design all the impossible states such as `verified=true` and `maybeDecoded=undefined`. 

No strong opinion, but if we can I would always use union types to model all and only the possible cases, where we can :) You can then pass this resulting type to other functions instead of maybeDecoded